source/common/upstream/load_balancer_impl.h

```diff
@@ -442,7 +480,13 @@ class RoundRobinLoadBalancer : public EdfLoadBalancerBase {
     // index.
     peekahead_index_ = 0;
   }
-  double hostWeight(const Host& host) override { return host.weight(); }
+  double hostWeight(const Host& host) override {
+    if (slow_start_enabled_) {
+      return applySlowStartFactor(host.weight(), host);
+    }
+    return host.weight();
+  }
```

Comment:
In the current implementation, the refresh method is not actively called after all nodes have left slow start mode. So even if all the final weights are the same, we will continue to use the EDF scheduler and keep on calculating the time factor which is already meaningless. 

In addition, assuming that the endpoints themselves have different weights, the EDF scheduler will always be used. Then after the slow start mode is over, the time factor calculation will continue even if the refresh method is actively called, which is also an unnecessary overhead. Alternatively, we can add an extra status bit to the endpoint to stop the time factor calculation after the node leaves the slow start mode.